articles/serializers.py

```diff
@@ -40,10 +48,183 @@ class Meta:
             "created_at",
             "updated_at",
             "author",
-            "favoritesCount",
+            "likes",
+            "dislikes",
         ]
 
+    def get_likes_count(self, instance: Any) -> Any:
+        return instance.likes.count()
+
+    def get_dislikes_count(self, instance: Any) -> Any:
+        return instance.dislikes.count()
+
     def create(self, validated_data: Any) -> Any:
         """set current user as author"""
-        validated_data["author"] = self.context.get("request").user  # type: ignore[union-attr]
+        validated_data["author"] = self.context.get("request").user
         return super().create(validated_data)
+
+    def to_representation(self, instance: Any) -> Any:
+
+        """check if the user liked or disliked an article and return favorited and unfavorited status"""
+        request = self.context.get("request")
+        representation = super().to_representation(instance)
+        if request.user.is_authenticated:
+            if request.user in instance.likes.all():
+                return {
+                    **representation,
+                    "favorited": True,
+                    "unfavorited": False,
+                }
+            if request.user in instance.dislikes.all():
+                return {
+                    **representation,
+                    "favorited": False,
+                    "unfavorited": True,
+                }
+            return {**representation, "favorited": False, "unfavorited": False}
+        return representation
+
+
+class ArticleFavoriteSerializer(serializers.ModelSerializer):
+    likes = UserSerializer(many=True, required=False, read_only=True)
+    dislikes = UserSerializer(many=True, required=False, read_only=True)
+    tags = TagListSerializerField()
+    likes_count = serializers.SerializerMethodField()
+    dislikes_count = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Article
+        fields = (
+            "lookup_id",
+            "slug",
+            "title",
+            "description",
+            "image",
+            "body",
+            "tags",
+            "is_hidden",
+            "likes_count",
+            "dislikes_count",
+            "likes",
+            "dislikes",
+            "created_at",
+            "updated_at",
+            "author",
+        )
+        read_only_fields = [
```

Comment:
does this mean fields like image, title. description are editable here?

Reply:
Corrected